shamir/shamir.go.commented

```diff
@@ -0,0 +1,237 @@
+package shamir
```

Comment:
Why the separate `.commented` file? Can't this be merged directly with shamir.go?

Reply:
@autrilla ^^^

Reply:
Oops, missed this. I'm not sure if we're allowed to modify the shamir file, and I removed some of their comments anyway. We could theoretically merge this, but idk if having an algorithm explanation in the source code alongside implementation details is a lot of noise. Your call.

Reply:
I don't think we should have a separate .commented file. I'd prefer to merge it alongside the source code.